### PR TITLE
[feature] add sms config

### PR DESCRIPTION
**Pull request by Aias00**
## What's changed?

add sms config in web app #1020 & backend 

which supports two types of sms providers(tencent\alibaba) for selecting, but i just achieved tencent sms, cause i found no alibaba`s config in code .  

## Checklist

- [x]  I have read the [Contributing Guide](https://hertzbeat.apache.org/docs/community/code_style_and_quality_guide)
- [x]  I have written the necessary doc or comment.
- [x]  I have added the necessary unit tests and all cases have passed.

## Add or update API

- [x] I have added the necessary [e2e tests](https://github.com/apache/hertzbeat/tree/master/e2e) and all cases have passed.

### CODE REVIEW

File: common/src/main/java/org/apache/hertzbeat/common/constants/SmsTypeEnum.java [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+     * get type
+     */
+    public static SmsTypeEnum getTypeByName(String type) {
+        for (SmsTypeEnum smsTypeEnum : values()) {
+            if (smsTypeEnum.name().equals(type)) {
+                return smsTypeEnum;
+            }
+
+        }
+        return null;
```
Comment (Calvin979) [inline common/src/main/java/org/apache/hertzbeat/common/constants/SmsTypeEnum.java:45]: Suggest use `stream().anyMatch()` to simplify code

File: common/src/main/java/org/apache/hertzbeat/common/constants/SmsTypeEnum.java [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+/**
+ * Sms type Enum
+ */
+public enum SmsTypeEnum {
+
+    /**
+     * tencent sms
+     */
+    tencent,
```
Comment (Calvin979) [inline common/src/main/java/org/apache/hertzbeat/common/constants/SmsTypeEnum.java:28]: Field in Enum should be uppercased

**Comment by tomsun28** [comment]
> which supports two types of sms providers(tencent\alibaba) for selecting, but i just achieved tencent sms, cause i found no alibaba`s config in code .

hi, yes there is no alibaba sms code, due the alibab sdk import too much irrelevant third-party dependencies. We plan to remove the SDKs of these cloud vendors and use the rest api to implement it

**Comment by Aias00** [comment]
hi，how about i remove backend’s modify, just keep web hi

---- Replied Message ----
| From | ***@***.***> |
| Date | 07/29/2024 12:25 |
| To | apache/hertzbeat ***@***.***> |
| Cc | aias00 ***@***.***>,
Author ***@***.***> |
| Subject | Re: [apache/hertzbeat] [feature] add sms config (PR #2399) |

which supports two types of sms providers(tencent\alibaba) for selecting, but i just achieved tencent sms, cause i found no alibaba`s config in code .

hi, yes there is no alibaba sms code, due the alibab sdk import too much irrelevant third-party dependencies. We plan to remove the SDKs of these cloud vendors and use the rest api to implement it

**Comment by tomsun28** [comment]
> hi，how about i remove backend’s modification, just keep web ui

ok 👍 suggest also remove sms config doc here https://hertzbeat.apache.org/docs/start/custom-config

**Comment by Aias00** [comment]
> > hi，how about i remove backend’s modification, just keep web ui
> 
> ok 👍 suggest also remove sms config doc here https://hertzbeat.apache.org/docs/start/custom-config

okay, cool

File: common/src/main/java/org/apache/hertzbeat/common/constants/SmsTypeEnum.java (hunk 2 above)
Comment (Aias00) [reply common/src/main/java/org/apache/hertzbeat/common/constants/SmsTypeEnum.java]: removed

**Comment by Aias00** [comment]
image
image
image
image

**Comment by Aias00** [comment]
now people can modify sms config by web ui , and the config can be persistenced. that`s all

**Review by tomsun28 — APPROVED** [review]
lgtm!

**Pull request merged**